cpp/pipelines/add_authority_external_ref.cc

```diff
@@ -130,9 +183,146 @@ void ParseGndWikidataMappingFile(std::string filename,
         LOG_ERROR("input or output files could not be opened");
 }
 
+struct GNDStructure {
+    std::string gnd_id, wikidata_personal_entity_id, wikipedia_personal_address;
+};
+
+bool IsThisCloseBracketForId(const std::string &url) {
+    int url_length = url.length();
+    std::string is_about(url.substr(url_length - 5, 5));
+    if (is_about.compare("about") == 0)
+        return true;
+
+    return false;
+}
+
+bool DoesItMatch(const std::string &url_based, const std::string &url_comp) {
+    const int base_string_lenght = url_based.length();
+    const std::string sub_string_url_comp = url_comp.substr(0, base_string_lenght);
+    if (url_based.compare(sub_string_url_comp) == 0)
+        return true;
+
+    return false;
+}
+
+int GenerateGNDAuthorityExternalRef(char *argv[]) {
+    auto load_file_start = std::chrono::high_resolution_clock::now();
+    std::ifstream input_file(argv[2]);
+    std::ofstream summary_file(argv[4]);
+
+    if (!input_file.is_open()) {
+        LOG_ERROR("can't open input file");
+        return 1;
+    }
+
+    const auto csv_file(FileUtil::OpenOutputFileOrDie(argv[3]));
+
+
+    const std::string d_nb_address("https://d-nb.info/gnd/");
```

Comment:
You can write it as `dnb_address`, there is no necessity for an additional underscore.